satpy/readers/nwcsaf_nc.py

```diff
@@ -217,7 +223,13 @@ def _mask_variable(variable):
 
     def _prepare_variable_for_palette(self, variable, info):
         try:
-            so_dataset = self.nc[self.file_key_prefix + info['scale_offset_dataset']]
+            if not isinstance(info['scale_offset_dataset'], list):
+                so_dataset = self.nc[self.file_key_prefix + info['scale_offset_dataset']]
+            else:
+                try:
+                    so_dataset = self.nc[self.file_key_prefix + info['scale_offset_dataset'][0]]
+                except KeyError:
+                    so_dataset = self.nc[self.file_key_prefix + info['scale_offset_dataset'][1]]
```

Comment:
This looks very similar to the code above, could these lines be factorised together somehow?